examples/phosh-demo/configuration.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+
+let
+  terminal = pkgs.kgx.override { genericBranding = true; };
+
+  # One-stop shop to customize the default username before building.
+  defaultUserName = "alice";
+in
+{
+  config = lib.mkMerge [
+    # INSECURE STUFF FIRST
+    # Users and hardcoded passwords.
+    {
+      # Forcibly set a password on users...
+      # Note that a numeric password is currently required to unlock a session
+      # with the plasma mobile shell :/
+      users.users.${defaultUserName} = {
+        isNormalUser = true;
+        # Numeric pin makes it **possible** to input on the lockscreen.
+        password = "1234";
+        home = "/home/${defaultUserName}";
+        extraGroups = [ "wheel" "networkmanager" "video" "feedbackd" ];
+        uid = 1000;
+      };
+
+      users.users.root.password  = "nixos";
+
+      # FIXME: highly insecure!
+      # FIXME: Figure out why this breaks...
+      #services.openssh.extraConfig = "PermitEmptyPasswords yes";
+    }
+
+    # "Desktop" environment configuration
+    {
+      powerManagement.enable = true;
+      hardware.opengl.enable = true;
+
+      systemd.defaultUnit = "graphical.target";
+      systemd.services.phosh = {
+        wantedBy = [ "graphical.target" ];
+        serviceConfig = {
+          ExecStart = "${pkgs.phosh}/bin/phosh";
+          User = 1000;
+          PAMName = "login";
+          WorkingDirectory = "~";
+
+          TTYPath = "/dev/tty7";
+          TTYReset = "yes";
+          TTYVHangup = "yes";
+          TTYVTDisallocate = "yes";
+
+          StandardInput = "tty-fail";
+          StandardOutput = "journal";
+          StandardError = "journal";
+
+          UtmpIdentifier = "tty7";
+          UtmpMode = "user";
+
+          Restart = "always";
+        };
+      };
+
+      services.xserver.desktopManager.gnome3.enable = true;
+
+      # Unpatched gnome-initial-setup is partially broken in small screens
+      services.gnome3.gnome-initial-setup.enable = false;
+
+      programs.phosh.enable = true;
+      environment.systemPackages = [ terminal ];
+      environment.gnome3.excludePackages = with pkgs.gnome3; [
+        gnome-terminal
+      ];
+
+      environment.etc."machine-info".text = lib.mkDefault ''
+        CHASSIS="handset"
+      '';
```

Comment:
Shouldn't most of this be part of a configuration option in upstream NixOS?

Relatedly, maybe, can this be done through a display manager?

Reply:
Valid values:

 - https://source.puri.sm/Librem5/phosh/-/blob/643e3b37d3c03f3840402d018924f77780b774e6/src/mode-manager.c

Reply:
hostnamectl reads /etc/machine-info, but only if it cannot find the chassis type via another method such as SMBIOS (?) or the device tree.

Fortunately, https://git.kernel.org/pub/scm/linux/kernel/git/sunxi/linux.git/commit/?h=sunxi/for-next&id=d900a1cd310de097fb94796ca575c118b2637b3b adds the chassis type for Pinephones. We just have to wait until the sunxi maintainers merge that into mainline.

Reply:
Or we can `fetchpatch` it / add it as needed.

Reply:
> Shouldn't most of this be part of a configuration option in upstream NixOS?

See https://github.com/NixOS/nixpkgs/pull/153940 for a PR to add the Phosh systemd service.